packages/block-library/src/link-preview/edit.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * WordPress dependencies
+ */
+import { __, _x } from '@wordpress/i18n';
+import { useState, useEffect } from '@wordpress/element';
+import { useBlockProps, BlockControls } from '@wordpress/block-editor';
+import { __experimentalFetchUrlData } from '@wordpress/core-data';
+import {
+	Placeholder,
+	Spinner,
+	Button,
+	ToolbarGroup,
+	ToolbarButton,
+} from '@wordpress/components';
+import { link, edit } from '@wordpress/icons';
+
+export default function LinkPreviewEdit( props ) {
+	const {
+		attributes: { url, title, icon, image },
+		setAttributes,
+	} = props;
+	const [ isFetching, setIsFetching ] = useState( false );
+	const [ isEditingUrl, setIsEditingUrl ] = useState( ! url );
+	const [ urlValue, setURLValue ] = useState( '' );
+
+	const blockProps = useBlockProps( {
+		href: url,
+		className: image ? 'has-image' : undefined,
+		onClick: isEditingUrl
+			? undefined
+			: ( event ) => {
+					event.preventDefault();
+			  },
+	} );
+
+	useEffect( () => {
+		if ( url && ! title && ! icon && ! image ) {
+			setIsFetching( true );
+			__experimentalFetchUrlData( url )
+				.then( ( data ) => {
+					setAttributes( data );
+				} )
+				.finally( () => {
+					setIsFetching( false );
+				} );
+		}
+	}, [] );
+
+	if ( isEditingUrl ) {
+		return (
+			<div { ...blockProps }>
+				<Placeholder
+					icon={ link }
+					label={ __( 'URL' ) }
+					instructions={ __(
+						'Paste a link to the content you want to display on your site.'
+					) }
+				>
+					<form
+						onSubmit={ () => {
+							setAttributes( {
+								url: urlValue,
+								title: '',
+								icon: '',
+								image: '',
+								description: '',
+							} );
+							setIsFetching( true );
+							__experimentalFetchUrlData( urlValue )
+								.then( ( data ) => {
+									setAttributes( data );
+									setIsEditingUrl( false );
+								} )
+								.finally( () => {
+									setIsFetching( false );
+								} );
+						} }
+					>
+						<input
+							type="url"
+							value={ urlValue }
+							className="components-placeholder__input"
+							aria-label={ __( 'URL' ) }
+							placeholder={ __( 'Enter URL to embed here…' ) }
+							onChange={ ( event ) => {
+								setURLValue( event.target.value );
+							} }
+						/>
```

Comment:
```suggestion
<InputControl
	label={ __( 'URL' ) }
	hideLabelFromVision
	className="components-placeholder__input"
	type="url"
	value={ urlValue }
	placeholder={ __( 'Enter URL to embed here…' ) }
	onChange={ ( event ) => {
		setURLValue( event.target.value );
	} }
/>
```

We can use the component here if we want to inherit the standard components package styling.

Reply:
It does have normal styling right now. Why are other blocks using `input` over `InputControl`? I just copied this from the embed block.

Reply:
Mainly just noticing that normalising towards our component library will ensure we have standard styling across all blocks. 

Not a blocker.